src/auth/auth.ts

```diff
@@ -599,11 +603,126 @@ class BaseAuth {
 }
 
 
+/**
+ * The tenant aware Auth class.
+ */
+export class TenantAwareAuth extends BaseAuth {
+  public readonly tenantId: string;
+
+  /**
+   * The TenantAwareAuth class constructor.
+   *
+   * @param {Auth} auth The Auth instance that created this tenant.
+   * @param tenantId The corresponding tenant ID.
+   * @constructor
+   */
+  constructor(private readonly auth: Auth, tenantId: string) {
+    super(
+        utils.getProjectId(auth.app),
+        new TenantAwareAuthRequestHandler(auth.app, tenantId),
+        cryptoSignerFromApp(auth.app));
+    utils.addReadonlyGetter(this, 'tenantId', tenantId);
```

Comment:
Just mark the constructor argument as `public readonly`

Reply:
One problem with this is that if they are not using typescript, they can overwrite this without any error thrown. I tried it and the property can be overwritten. Since this is publicly available, i think we should enforce the readonly.

Reply:
I expected the TS compiler to generate a readonly getter for this, but looks like that only happens if we expose this as a getter in the code. So this is ok.

Reply:
I expected the TS compiler to add a readonly getter, but looks like that only happens if we define a getter explicitly here. So this is ok.